faculty/clients/auth.py

```diff
@@ -104,3 +105,12 @@ def __call__(self, request):
         request.headers["Authorization"] = header_content
 
         return request
+
+
+class SherlockMLAuth(FacultyAuth):
```

Comment:
I'm not sure that we want to have this alias here at all - if users are already migrating from using the `sherlockml` library to `faculty`, it seems strange to allow them to leave this `Auth` class as `SherlockMLAuth` and not just force them to update that as well.

Spoke with @pbugnion about this and he thinks there's some value in allowing people to upgrade gradually. Any thoughts @zblz @srstevenson ?

Reply:
I side with @acroz: I favour replacing all references to SherlockML with references to Faculty atomically. We're changing the package name and import paths, so users will already need to change their imports and can change the class name at the same time. A `grep -i` for `sherlockml` will allow them to ensure they've not missed anything.

Reply:
I think `SherlockMLAuth(FacultyAuth)` should only be in the `sherlockml` library during the deprecation period. As @acroz and @srstevenson said, the `faculty` library should be free of SherlockML references (except maybe using old env vars and paths for credentials)

Reply:
Fair enough. I'll remove it.